generator/template.go

```diff
@@ -229,20 +347,39 @@ func fieldName(r *registry.Registry) func(name string) string {
 func renderURL(r *registry.Registry) func(method data.Method) string {
 	fieldNameFn := fieldName(r)
 	return func(method data.Method) string {
-		url := method.URL
+		methodURL := method.URL
 		reg := regexp.MustCompile("{([^}]+)}")
-		matches := reg.FindAllStringSubmatch(url, -1)
+		matches := reg.FindAllStringSubmatch(methodURL, -1)
+		fieldsInPath := "["
 		if len(matches) > 0 {
 			log.Debugf("url matches %v", matches)
 			for _, m := range matches {
 				expToReplace := m[0]
 				fieldName := fieldNameFn(m[1])
 				part := fmt.Sprintf(`${req["%s"]}`, fieldName)
-				url = strings.ReplaceAll(url, expToReplace, part)
+				methodURL = strings.ReplaceAll(methodURL, expToReplace, part)
+				fieldsInPath += fmt.Sprintf(`"%s", `, fieldName)
+			}
+		}
+		fieldsInPath = strings.TrimRight(fieldsInPath, ", ") + "]"
```

Comment:
for constructing strings like an array like this `[a,b,c,d]`.

Would be better to use arrays and join construct the string rather than construct it and trim right. So you can do something like

```golang
fieldsInpath := make([]string, 0, len(matches))

// in the for loop
fieldsInPath := append(fieldsInPath, fieldName)

// out of the loop
fieldsInPath = fmt.Sprintf("[%s]", strings.Join(fieldsInPath, ", "))

```

Reply:
Ah! Sorry for my naive implementation. Will revise as per your suggestion 🙇🏼

Reply:
Hmm, I tried your example and unfortunately, the fieldsInPath gets generated like this `[a, b, c]` but the expected generated list should be like this `["a", "b", "c"]` 😞 and the tests fail. I am trying to figure out if we can use your suggestion to generate `["a", "b", "c"]` 🙏🏼

Reply:
Done 🙇🏼